File: term-frequency.java
import java.io.*;
import java.util.*;
import java.lang.*;

public class Solution {
    public static void main(String args[] ) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int wordLength = Integer.parseInt(br.readLine());

        ArrayList<String> words = new ArrayList<String>();
        for (int i = 0; i < wordLength; i++) {
            words.add(br.readLine());
        }

        HashMap<String, Integer> frequencies = new HashMap<String, Integer>();
        for (int i = 0; i < wordLength; i++) {
          Integer frequency = frequencies.get(words.get(i));
          if (frequency == null){
            frequency = 0;
          }
          frequencies.put(words.get(i), frequency+1);
        }

        ArrayList<String> sortedWords = new ArrayList<String>(frequencies.keySet());
        Collections.sort(sortedWords, new FrequencyCompare(frequencies));

        int outputWordLength = Integer.parseInt(br.readLine());
        for (int i = 0; i < outputWordLength; i++) {
          System.out.println(sortedWords.get(i));
        }
    }

    public static class FrequencyCompare implements Comparator<String> {
        private HashMap<String, Integer> frequencies;

        public FrequencyCompare(HashMap<String, Integer> frequencies){
          this.frequencies = frequencies;
        }

        @Override
        public int compare(String a, String b) {
            int frequency = this.frequencies.get(b).compareTo(this.frequencies.get(a));
            if (frequency == 0){
              return a.compareTo(b);
            } else{
              return frequency;
            }
        }
    }
}

File: term-frequency.js
process.stdin.resume();
process.stdin.setEncoding('utf8');

var words = [];
var outputWordLength = 0, wordLength = 0;
// Read the inputs
process.stdin.on('data', function(chunk) {
  var inputs = chunk.split("\n");
  wordLength = parseInt(inputs.shift());
  for (var i = 0; i < wordLength; i++){
    words.push(inputs.shift());
  }
  outputWordLength = parseInt(inputs.shift());
});

// After reading the inputs
process.stdin.on('end', function() {
  var frequencies = {};
  for (var i = 0; i < wordLength; i++){
    var word = words[i];
    frequencies[word] = frequencies[word] || 0;
    frequencies[word]++;
  }

  var sortedWords = Object.keys(frequencies).sort(function(a, b){
    var compareFrequency = frequencies[b] - frequencies[a];
    return compareFrequency == 0 ? a.localeCompare(b) : compareFrequency;
  });

  for(var i = 0; i < outputWordLength; i++){
    console.log(sortedWords[i]);
  }
});
